image/histogram_44312042/equalize_adapthist.py

```diff
@@ -0,0 +1,60 @@
+"""
+COMP3710 Assignment 3
+
+Author: James Copperthwaite
+
+Contrast Limited Adaptive Histogram Equalization (CLAHE).
+
+"""
+
+
+import tensorflow as tf
+import numpy as np
+
+
+def histogram(image, nbins=256, normalize=True):
+    sess = tf.InteractiveSession()
+    
+    img = tf.constant(image.astype(np.float32))
+    hist = tf.Variable(np.zeros(nbins).astype(np.int64))
+    out = tf.Variable(img)
+
+    tf.global_variables_initializer().run() #init variables
+
+    img = tf.reshape(img, [-1]) 
+
+    current_min = tf.dtypes.cast(tf.reduce_min(img), tf.float32)
+    current_max = tf.dtypes.cast(tf.reduce_max(img), tf.float32)
+    
+    bins = tf.linspace(current_min, current_max, nbins+1)
+    
+    y, idx, count = tf.unique_with_counts(img)
+
+    for i in range(nbins):
+        if i==nbins-1:
+            mask = (img <= bins[i+1]) 
+            lim = tf.boolean_mask(img, mask)
+
+        else:
+            mask = (img < bins[i+1]) 
```

Comment:
Good to add inline comments even if original code didn't have them

Reply:
Added in latest commit